apps/backoffice-v2/src/common/components/molecules/ProcessTracker/hooks/useProcessTrackerLogic/useProcessTrackerLogic.tsx

```diff
@@ -1,127 +1,61 @@
-      }
+  const [uncollapsedItemValue, setUncollapsedItemValue] = useState<string>();
+  const onValueChange = useCallback((value: string) => {
+    setUncollapsedItemValue(value);
+  }, []);
 
-      return processStatusToIcon[ProcessStatus.IDLE];
-    },
-    [context?.flowConfig?.stepsProgress],
-  );
-  const getPluginByName = useCallback(
-    (name: string) => {
-      let plugin: NonNullable<TWorkflowById['context']['pluginsOutput']>[string];
+  const processTrackers = useMemo(
+    () =>
+      processes.reduce((list, processName) => {
+        const ProcessTracker = processTrackersMap[processName as keyof typeof processTrackersMap];
 
-      Object.keys(context?.pluginsOutput ?? {})?.forEach(key => {
-        if (context?.pluginsOutput?.[key]?.name !== name) {
-          return;
+        if (!ProcessTracker) {
+          console.warn(`${processName} is unsupported.`);
+          return list;
         }
 
-        plugin = context?.pluginsOutput?.[key];
-      });
+        list.push(new ProcessTracker(workflow, plugins));
 
-      return plugin;
-    },
-    [context?.pluginsOutput],
+        return list;
+      }, [] as IProcessTracker[]),
+    [workflow, plugins, processes],
   );
-  const getUboFlowStatus = useCallback((tags: TWorkflowById['tags']) => {
-    const tag = tags?.find(tag => tagToIcon[tag as keyof typeof tagToIcon]);
-
-    if (!tag) {
-      return tagToIcon.DEFAULT;
-    }
-
-    return tagToIcon[tag as keyof typeof tagToIcon];
-  }, []);
-
-  const collectionFlowSubitems = useMemo(() => {
-    return steps?.map(step => {
-      return {
-        text: titleCase(step),
-        leftIcon: getCollectionFlowStatus(step),
-      };
-    });
-  }, [getCollectionFlowStatus, steps]);
-  const thirdPartyProcessesSubitems = useMemo(() => {
-    return plugins
-      ?.filter(({ name }) => pluginsWhiteList.includes(name as (typeof pluginsWhiteList)[number]))
-      ?.map(({ displayName, name }) => {
-        const pluginStatus = getPluginByName(name)?.status ?? ProcessStatus.DEFAULT;
 
-        return {
-          text:
-            pluginStatus === ProcessStatus.CANCELED ? (
-              <span className={`text-slate-400/40 line-through`}>{displayName}</span>
-            ) : (
-              displayName
-            ),
-          leftIcon: processStatusToIcon[pluginStatus as keyof typeof processStatusToIcon],
-        };
-      });
-  }, [getPluginByName, plugins]);
-  const uboFlowsSubitems = useMemo(() => {
-    return kycChildWorkflows?.map(({ context, tags }) => {
+  const trackedProcesses = useMemo(() => {
+    return processTrackers.map(processTracker => {
       return {
-        text: `${valueOrNA(context?.entity?.data?.firstName)} ${valueOrNA(
-          context?.entity?.data?.lastName,
-        )}`,
-        leftIcon: getUboFlowStatus(tags),
+        title: processTracker.getReadableName(),
+        name: processTracker.PROCESS_NAME,
+        subitems: processTracker.buildItems(),
```

Comment:
Not too sure if at least `getReadableName` has value in being a method and not just a property. And would make sense to have some reference to the word "title" in the prop/method.